runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/MemoryChunk.java

```diff
@@ -0,0 +1,811 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.nemo.runtime.executor.data;
+
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+
+/**
+ * This class represents chunk of memory residing in off-heap region
+ * managed by {@link MemoryPoolAssigner}, which is backed by {@link ByteBuffer}.
+ */
+public class MemoryChunk {
+
+  @SuppressWarnings("restriction")
+  protected static final sun.misc.Unsafe UNSAFE = getUnsafe();
+  @SuppressWarnings("restriction")
+  protected static final long BYTE_ARRAY_BASE_OFFSET = UNSAFE.arrayBaseOffset(byte[].class);
+  private static final boolean LITTLE_ENDIAN = (ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN);
+  private final ByteBuffer buffer;
+  private long address;
```

Comment:
Can you briefly comment on why we access and use address/addressLimit here? (Rather than simply using the ByteBuffer abstraction)
e.g.,correctness(sanity checks), time efficiency, space efficiency, ...